src/core_plugins/elasticsearch/index.js

```diff
@@ -15,8 +16,6 @@ const DEFAULT_REQUEST_HEADERS = [ 'authorization' ];
 
 module.exports = function ({ Plugin }) {
   return new Plugin({
-    require: ['kibana'],
```

Comment:
The `uiExports` collection is already available in plugins via the `PluginAPI` passed into each plguin's index.js file, so the elasticsearch plugin would do something like this:

```js
export default (kibana) => (
  new kibana.Plugin({
    init() {
      kibana.uiExports.savedObjectMappings.forEach(mapping => {
        someService.addMapping(mapping);
      });
    }
  })
)
```